src/scenes/PlantingSitesRouter/view/GenericSiteView.tsx

```diff
@@ -0,0 +1,193 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Box, Typography, Grid, Theme, useTheme, List, ListItem } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import { DateTime } from 'luxon';
+import { Button, DropdownItem } from '@terraware/web-components';
+import { useDeviceInfo } from '@terraware/web-components/utils';
+import TextField from '@terraware/web-components/components/Textfield/Textfield';
+import strings from 'src/strings';
+import { APP_PATHS } from 'src/constants';
+import isEnabled from 'src/features';
+import { PlantingSeason, SiteDetails } from 'src/types/Tracking';
+import { useProjects } from 'src/hooks/useProjects';
+import PageSnackbar from 'src/components/PageSnackbar';
+import BoundariesAndZones from './BoundariesAndZones';
+import BackToLink from 'src/components/common/BackToLink';
+import { useLocationTimeZone } from 'src/utils/useTimeZoneUtils';
+import Card from 'src/components/common/Card';
+import OptionsMenu from 'src/components/common/OptionsMenu';
+import { View } from 'src/components/common/ListMapSelector';
+import SimplePlantingSite from './SimplePlantingSite';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  titleWithButton: {
+    alignItems: 'center',
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    minHeight: '50px',
+  },
+}));
+
+export type GenericSiteViewProps<T extends SiteDetails> = {
+  editDisabled?: boolean;
+  editUrl: string;
+  onDelete: () => void;
+  plantingSite: T;
+};
+
+export default function GenericSiteView<T extends SiteDetails>({
```

Comment:
The bulk of this was extracted out from `PlantingSiteView`, no new functionality.